src/index.ts

```diff
@@ -93,38 +216,204 @@ export class ConsentControl {
      * @param {ITextResources} textResources the text want to be displayed
      */
     public setTextResources(textResources: ITextResources): void {
-        if (textResources.bannerMessageHtml) {
-            this.textResources.bannerMessageHtml = textResources.bannerMessageHtml;
+        for (let key of Object.keys(this.textResources)) {
+            let typeKey = <keyof ITextResources> key;
+
+            if (textResources[typeKey]) {
+                this.textResources[typeKey] = textResources[typeKey];
+            }
+        }
+    }
+
+    /**
+     * Set themes in the banner and preferences dialog
+     * If a theme is not included, the default one will be used
+     * 
+     * @param {IThemes} themes themes that will be used in the banner and preferences dialog
+     */
+    public setThemes(themes?: IThemes): void {
+        this.themes.light = this.defaultLightTheme;
+        if (themes?.light) {
+            this.setCertainTheme(themes.light, this.themes.light);
+        }
+
+        this.themes.dark = this.defaultDarkTheme;
+        if (themes?.dark) {
+            this.setCertainTheme(themes.dark, this.themes.dark);
         }
-        if (textResources.acceptAllLabel) {
-            this.textResources.acceptAllLabel = textResources.acceptAllLabel;
+
+        this.themes["high-contrast"] = this.defaultHighContrast;
+        if (themes?.["high-contrast"]) {
+            this.setCertainTheme(themes["high-contrast"], this.themes["high-contrast"]);
+        }
+    }
+
+    /**
+     * Use the passed theme to set the theme property
+     * 
+     * @param {ITheme} theme the passed theme that we want to display
+     * @param {ITheme} currentTheme the theme property that we want to set
+     */
+    public setCertainTheme(theme: ITheme, currentTheme: ITheme): void {
+        for (let key of Object.keys(currentTheme)) {
+            let typeKey = <keyof ITheme> key;
+
+            if (theme[typeKey]) {
+                currentTheme[typeKey] = theme[typeKey];
+            }
+        }
+
+        // Styles that can be determined by another one
+
+        if (theme["banner-background-color"] && !theme["dialog-background-color"]) {
+            currentTheme["dialog-background-color"] = theme["banner-background-color"];
+        }
+
+        if (theme["dialog-background-color"]) {
```

Comment:
That's really nice how you calc all the colors here.
Let's make some of the basic colors as required (like `dialog-background-color`, `primary-button-color`,...) in `ITheme` and all the ones which can be calculated - leave as optional.